go/libraries/doltcore/env/multi_repo_env.go

```diff
@@ -53,6 +54,74 @@ type MultiRepoEnv struct {
 	dialProvider dbfactory.GRPCDialProvider
 }
 
+type StorageMetadataMap map[string]nbs.StorageMetadata
+
+func (sms StorageMetadataMap) ArchiveFilesPresent() bool {
+	for _, sm := range sms {
+		if sm.ArchiveFilesPresent() {
+			return true
+		}
+	}
+	return false
+}
+
+func GetMultiEnvStorageMetadata(dataDirFS filesys.Filesys) (StorageMetadataMap, error) {
+
+	dbMap := make(map[string]filesys.Filesys)
+
+	path, err := dataDirFS.Abs("")
+	if err != nil {
+		return nil, err
+	}
+	envName := getRepoRootDir(path, string(os.PathSeparator))
+	dbName := dbfactory.DirToDBName(envName)
+	dbMap[dbName] = dataDirFS
+
+	// If there are other directories in the directory, try to load them as additional databases
+	dataDirFS.Iter(".", false, func(path string, _ int64, isDir bool) (stop bool) {
+		if !isDir {
+			return false
+		}
+
+		dir := filepath.Base(path)
+
+		newFs, er2 := dataDirFS.WithWorkingDir(dir)
+		if er2 != nil {
+			return false
+		}
+		path, er2 = newFs.Abs("")
+		if er2 != nil {
+			return false
+		}
+		envName := getRepoRootDir(path, string(os.PathSeparator))
+
+		falseEnv := IncompleteEnv(newFs)
+		if !falseEnv.Valid() {
+			return false
```

Comment:
In this particular case, where it's an incomplete env, the reasons it would be invalid are that it doesn't have a .dolt directory. This is basically to filter out things which don't look remotely like dolt databases. I'll add a comment.